picard/track.py

```diff
@@ -71,8 +71,23 @@ def add_file(self, file):
     def update_file_metadata(self, file):
         if file not in self.linked_files:
             return
-        file.copy_metadata(self.metadata)
+        file.copy_metadata(self.orig_metadata)
         file.metadata['~extension'] = file.orig_metadata['~extension']
+        self.metadata.copy(file.metadata)
+
+        # Re-run tagger scripts with updated metadata
+        if config.setting["enable_tagger_scripts"]:
+            for s_pos, s_name, s_enabled, s_text in config.setting["list_of_scripts"]:
+                if s_enabled and s_text:
```

Comment:
To reduce code redundancy i'd introduce a `enabled_tagger_scripts_list()`:

```python
def enabled_tagger_scripts_texts():
    if not config.setting["enable_tagger_scripts"]:
        return []
    return [ s_text for s_pos, s_name, s_enabled, s_text in config.setting["list_of_scripts"] if s_enabled and s_text]
```

So you can just do:

```python
for s_text in enabled_tagger_scripts_texts():
   parser = ScriptParser()
   try:
        parser.eval(s_text, file.metadata)
        parser.eval(s_text, self.metadata)
   except:
....
```

Reply:
Since pyt3, we can do this - 
```python
def enabled_tagger_scripts_texts():
    if not config.setting["enable_tagger_scripts"]:
        return []
    return [ s_text for *_discard, s_enabled, s_text in config.setting["list_of_scripts"] if s_enabled and s_text]
```

Reply:
I like this change, will apply. I'll put the new function into `picard/script.py`.